specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2021-01-01/kusto.json

```diff
@@ -0,0 +1,4925 @@
+          "description": "The tenant id of the principal"
+        },
+        "principalType": {
+          "description": "Principal type.",
+          "type": "string",
+          "enum": [
+            "App",
+            "Group",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "PrincipalType",
+            "modelAsString": true
+          }
+        },
+        "tenantName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The tenant name of the principal"
+        },
+        "principalName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The principal name"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The provisioned state of the resource."
+        }
+      },
+      "description": "A class representing database principal property."
+    },
+    "ClusterPrincipalProperties": {
+      "type": "object",
+      "required": [
+        "principalId",
+        "role",
+        "principalType"
+      ],
+      "properties": {
+        "principalId": {
+          "description": "The principal ID assigned to the cluster principal. It can be a user email, application ID, or security group name.",
+          "type": "string"
+        },
+        "role": {
+          "description": "Cluster principal role.",
+          "type": "string",
+          "enum": [
+            "AllDatabasesAdmin",
+            "AllDatabasesViewer"
+          ],
+          "x-ms-enum": {
+            "name": "ClusterPrincipalRole",
+            "modelAsString": true
+          }
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "The tenant id of the principal"
+        },
+        "principalType": {
+          "description": "Principal type.",
+          "type": "string",
+          "enum": [
+            "App",
+            "Group",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "PrincipalType",
+            "modelAsString": true
+          }
+        },
+        "tenantName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The tenant name of the principal"
+        },
+        "principalName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The principal name"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The provisioned state of the resource."
+        }
+      },
+      "description": "A class representing cluster principal property."
+    },
+    "ScriptProperties": {
+      "type": "object",
+      "required": [
+        "scriptUrl",
+        "scriptUrlSasToken"
+      ],
+      "properties": {
+        "scriptUrl": {
+          "description": "The url to the KQL script blob file.",
+          "type": "string"
+        },
+        "scriptUrlSasToken": {
```

Comment:
set x-ms-mutability to "create" as well (+ update if you let people change this).  It should never be returned in a GET response or as a response to the PUT.